### ISSUE TITLE
Bytes vector compiles without termination condition for bytes field.

### ISSUE THREAD

**Issue by sethhall** [zeek/spicy#1832]
This toy parser is applying size to the total size of the vector but the bytes in the vector actually have no termination condition.

```spicy
module test;

public type X = unit {
    a: bytes[10] &size=5;
};
```

**Comment by rsmmr** [comment]
Looks like there's validation missing rejecting the inner field: it should  use the same logic that we use for `bytes` fields outside of vectors to enforce that there's some termination condition (e.g, `&eod`, `&size`, etc.).

**Comment by bbannier** [comment]
Fixed with f9bbf82b7cadbf31f0e606708c380f42022847cc.
